### PR TITLE
test: Using the same step resources during tests execution as in final binary

### LINKED ISSUE #815: Not using `main` branches for camel-component-metadata and kaoto-viewdefinition-catalog during tests
I noticed that now some application.yaml contains git url for resources (camel-component-metadata, kamelets etc.) which points to the main branch however the application.yaml which is used for prod profile uses only files from resources.

Due to that, during JVM testing, the Kaoto-backend is tested with different resources as used in production. (other words, JVM and Native testing doesn't use same resources). 

Also, when something changed in main branch in that repository, the change is already presented in kaoto-backend tests which can bring a failure of the tests (which relates to that resources, e.g. StepResourceTest https://github.com/KaotoIO/kaoto-backend/pull/812 ) to the any PR. That can be hard to debug/find a root cause because the step which is not related to the PR start failing. 

From my POV, the tests should be running against the same resources as they are used in the final binary.

Two possible solutions we can have:
- Instead of `main` branch, we can use commit hash. And when the camel-component-metadata file is updated in kaoto-backend, also the commit hash will be updated.
- Instead of git, we can use the same resources as it is used in prod profile. And we can create test which will check if `camel-component-metadata` file has same steps as in the main, and if not, the test failed with a specific message what is wrong.  ( same as here: https://github.com/KaotoIO/kaoto-backend/blob/main/camel-support/src/test/java/io/kaoto/backend/camel/metadata/parser/step/kamelet/KameletParseCatalogTest.java#L192)
- Making releases for camel-component-metadata and kaoto-viewdefinition-catalog

### PR DESCRIPTION
**Pull request by mkralik3**
Commits:
- Changed remote branches to file resources in tests configurations in order to use the same resources during the test as they are used in final binary
- Created script for easiest step catalogs (zips) update `update-resources.sh`
- Added version info into Readme
- Updated resources by update script to have all resources in line with declared version (e.g. camel-yaml-dsl.json was not the same as it is in 3.21.0)

### CODE REVIEW

File: camel-support/src/test/resources/application.yaml [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
-    git:
-      -
-        url: "https://github.com/KaotoIO/camel-component-metadata.git"
-        tag: "main"
+      - url: "resource://camel-kamelets-3.21.0.jar"
+      - url: "resource://camel-connectors-3.21.0.zip"
+      - url: "resource://camel-component-metadata.zip"
   viewdefinition:
     jar:
-      - url: "https://github.com/KaotoIO/kaoto-viewdefinition-catalog/archive/refs/tags/test.zip"
```
Comment (Delawen) [inline camel-support/src/test/resources/application.yaml:12]: Are we testing the remote url retrieval of resources somewhere else?

Comment (mkralik3) [reply camel-support/src/test/resources/application.yaml:12]: I found it only for Kamelet, https://github.com/KaotoIO/kaoto-backend/blob/main/camel-support/src/test/java/io/kaoto/backend/camel/metadata/parser/step/kamelet/KameletParseCatalogTest.java#L71C33-L71C33

I can add something similar for each type of resource.
Or we can add another profile which will test `api` package with remote types of resources.

Comment (mkralik3) [reply camel-support/src/test/resources/application.yaml:12]: I am working on that tests

Comment (mkralik3) [reply camel-support/src/test/resources/application.yaml:12]: @Delawen Added tests for that https://github.com/KaotoIO/kaoto-backend/pull/818/commits/328b72fc4accbd1f8c483a1440083193a7703d2c

**Review by Delawen — APPROVED** [review]

**Pull request merged**